src/deluge/processing/stem_export/stem_export.cpp

```diff
@@ -427,84 +427,49 @@ void StemExport::displayStemExportProgress7SEG() {
 	display->setText(exportStatus.c_str(), true, 255, false);
 }
 
-/// based on Stem Export Type, will set a WAV file name in the format of:
-/// /OutputType_StemExportType_OutputName_IndexNumber.WAV
-/// example: /SYNTH_CLIP_BASS SYNTH_00000.WAV
-/// example: /SYNTH_TRACK_BASS SYNTH_00000.WAV
-/// this wavFileName is then concatenate to the filePath name to export the WAV file
-void StemExport::setWavFileNameForStemExport(StemExportType stemExportType, Output* output, int32_t fileNumber) {
-	// wavFileNameForStemExport = "/"
-	Error error = wavFileNameForStemExport.set("/");
-	if (error != Error::NONE) {
-		return;
-	}
+// creates the full file path for stem exporting including the stem folder structure and wav file name
+Error StemExport::getUnusedStemRecordingFilePath(String* filePath, AudioRecordingFolder folder) {
+	const auto folderID = util::to_underlying(folder);
 
-	// wavFileNameForStemExport = "/OutputType
-	const char* outputType;
-	switch (output->type) {
-	case OutputType::AUDIO:
-		outputType = "AUDIO";
-		break;
-	case OutputType::SYNTH:
-		outputType = "SYNTH";
-		break;
-	case OutputType::KIT:
-		outputType = "KIT";
-		break;
-	default:
-		break;
-	}
-	error = wavFileNameForStemExport.concatenate(outputType);
+	Error error = storageManager.initSD();
 	if (error != Error::NONE) {
-		return;
+		return error;
 	}
 
-	// wavFileNameForStemExport = "/OutputType_
-	error = wavFileNameForStemExport.concatenate("_");
+	error = stemExport.getUnusedStemRecordingFolderPath(filePath, folder);
```

Comment:
should just be getUnusedStemRecordingFolderPath and not refer to a global object